plugins/include/nd_stocks.inc

```
@@ -0,0 +1,267 @@
+ */
+
+stock void ND_SetPlayerClass(int client, ND_MainClasses MainClass, int SubClass, int gizmo = ND_GIZMO_NONE)
+{
+	SetEntProp(client, Prop_Send, "m_iPlayerClass", view_as<int>MainClass);
+    	SetEntProp(client, Prop_Send, "m_iPlayerSubclass", SubClass);
+	SetEntProp(client, Prop_Send, "m_iDesiredPlayerClass", view_as<int>MainClass);
+	SetEntProp(client, Prop_Send, "m_iDesiredPlayerSubclass", SubClass);
+	SetEntProp(client, Prop_Send, "m_iDesiredGizmo", gizmo);
+}
+
+/** 
+ * Get the main class of a client index
+ *
+ * @param client		Client index.
+ * @return			The main class of the client, see mainclass enum
+ */
+
+stock ND_MainClasses ND_GetPlayerMainClass(int client)
+{
+	return view_as<ND_MainClasses>(GetEntProp(client, Prop_Send, "m_iPlayerClass"));
+}
+
+/** 
+ * Get the sub class of a client index
+ *
+ * @param client		Client index.
+ * @return			The sub class of the client, see subclass consants
+ */
+
+stock int ND_GetPlayerSubClass(int client)
+{
+	return GetEntProp(client, Prop_Send, "m_iPlayerSubclass");
+}
+
+
+/* Commander Functions */
+
+/** 
+ * Get wetheir or not a client is commander
+ *
+ * @param client		Client index.
+ * @return			True if commander otherwise false
+ * @error			Invalid client index or client not ingame
+ */
+stock bool ND_IsCommander(client)
+{
+	return ND_GetTeamCommander(GetClientNDTeam(client)) == client;
+}
+
+/** 
+ * Get the commander belonging to a particular team
+ *
+ * @param team			NDTeam enum.
+ * @return			The client index
+ * @error			Invalid team or -1 if no commander
+ */
+stock int ND_GetTeamCommander(NDTeam team)
+{
+	return GameRules_GetPropEnt("m_hCommanders", view_as<int>team - 2);
+}
+
+/** 
+ * Demote a commander on a specified team
+ *
+ * @param team			NDTeam enum.
+ * @error			Invalid team or no commander
+ */
+
+stock void ND_DemoteTeamCommander(NDTeam team)
+{
+	int commander = ND_GetTeamCommander(team);
+	
+	FakeClientCommand(commander, "startmutiny");
```

An IsValidCommander() function would be better. Raising an error is a good thing in this case instead of just having it silently fail, so, the client knows they've done something wrong.